tests/test_control.py

```diff
@@ -402,28 +402,72 @@ def process(self, instance):
     pyblish.api.register_plugin(Validate)
 
     c = reset()
-    
+
     def has_enabled_validator(c):
         """Return whether Validate validator is enabled/compatible"""
-        
+
         rows = c.plugin_proxy.rowCount()
         for row in range(rows):
             item = c.plugin_proxy.item(row)
             if item.name == 'Validate':
                 return True
-                
+
         return False
-       
+
     item = c.item_model.instances["A"]
     index = c.item_model.items.index(item)
-        
+
     # Default state (enabled)
     assert has_enabled_validator(c), "Not enabled before"
-    
+
     # toggle off
     c.toggleInstance(index)
     assert not has_enabled_validator(c), "Not disabled"
-    
-    # toggle back on    
+
+    # toggle back on
     c.toggleInstance(index)
     assert has_enabled_validator(c), "Not enabled after"
+
+
+@with_setup(lib.clean)
+def test_action_availability():
+    """failed plugins with failure action, needs to have an action"""
+
+    class SelectMany(pyblish.api.ContextPlugin):
+        order = pyblish.api.CollectorOrder
+
+        def process(self, context):
+            for name in ("A", "B", "C"):
+                instance = context.create_instance(name)
+                instance.set_data("family", "myFamily")
+
+    class FailureAction(pyblish.api.Action):
+        on = "failed"
+
+    class ValidateFailA(pyblish.api.InstancePlugin):
+        order = pyblish.api.ValidatorOrder
+        families = ["myFamily"]
+        actions = [FailureAction]
+
+        def process(self, instance):
+            assert str(instance) != "A"
+
+    class ValidateFailBC(pyblish.api.InstancePlugin):
+        order = pyblish.api.ValidatorOrder
+        families = ["myFamily"]
+        actions = [FailureAction]
+
+        def process(self, instance):
+            assert str(instance) == "A"
+
+    pyblish.api.deregister_all_plugins()
+    pyblish.api.register_plugin(SelectMany)
+    pyblish.api.register_plugin(ValidateFailA)
+    pyblish.api.register_plugin(ValidateFailBC)
+
+    c = reset()
+
+    validate(c)
+
+    assert c.item_model.plugins["ValidateFailA"].actions
```

Comment:
It passed because it would have passed regardless of a plug-in failing.

That list contains all actions of the plug-in, not the currently available actions in the GUI, that's what the [`getPluginActions()`](https://github.com/pyblish/pyblish-qml/blob/29c24ddbafaf2b6edc365ee3027596c580dd0e08/pyblish_qml/control.py#L302) method is for. That will take into account whether anything has failed or not.

I modified your test, and added a plug-in that did not fail. Have a look and see if it makes sense.

- https://github.com/pyblish/pyblish-qml/blob/master/tests/test_control.py#L487